drv/stm32h7-sprot-server/src/main.rs

```diff
@@ -522,59 +497,31 @@ impl ServerImpl {
         rsp: MsgType,
         timeout: u32,
         attempts: u16,
-    ) -> Result<Option<u32>, RequestError<UpdateError>> {
-        let size =
-            compose(req, payload_len, &mut self.tx_buf[..]).map_err(|_| {
-                idol_runtime::RequestError::Runtime(UpdateError::SpRotError)
-            })?;
-        ringbuf_entry!(Trace::TxSize(size));
-        let (msgtype, payload_len) = self
-            .do_send_recv_retries(size, timeout, attempts)
-            .map_err(|_| {
-                idol_runtime::RequestError::Runtime(UpdateError::SpRotError)
-            })?;
-        if msgtype == rsp {
-            let buf = payload_buf(Some(payload_len), &self.rx_buf[..]);
-            let (rsp, _) = hubpack::deserialize::<UpdateRspHeader>(buf)
-                .map_err(|_| {
-                    idol_runtime::RequestError::Runtime(UpdateError::SpRotError)
-                })?;
+    ) -> Result<Option<u32>, SprotError> {
+        let txmsg = self.tx_buf.from_existing(req, payload_len)?;
+        ringbuf_entry!(Trace::TxSize(txmsg.0));
+        let rxmsg @ VerifiedRxMsg(header) =
```

Comment:
Same question here: could we dodge the cursed syntax and use `rxmsg.header`?

Reply:
yup

Reply:
Fixed in [3a0f61a](https://github.com/oxidecomputer/hubris/pull/965/commits/3a0f61ac0031aaed46524561bf56eb41f20f085f)